catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -0,0 +1,186 @@
+      LOGGER.debug("Error fetching cql request, empty or invalid body.");
+      response.status(HttpStatus.BAD_REQUEST_400);
+      return ImmutableMap.of("message", "Bad request");
+    }
+
+    LOGGER.trace("Finding transformer to transform query response.");
+
+    ServiceReference<QueryResponseTransformer> queryResponseTransformer =
+        queryResponseTransformers
+            .stream()
+            .filter(transformer -> transformer.getProperty("id").equals(transformerId))
+            .findFirst()
+            .orElse(null);
+
+    if (queryResponseTransformer == null) {
+      LOGGER.debug("Could not find transformer with id: {}", transformerId);
+      response.status(HttpStatus.NOT_FOUND_404);
+      return ImmutableMap.of("message", "Service not found");
+    }
+
+    CqlQueryResponse cqlQueryResponse = util.executeCqlQuery(cqlRequest);
+
+    attachFileToResponse(request, response, queryResponseTransformer, cqlQueryResponse);
+
+    return "";
+  }
+
+  private void setHttpHeaders(Request request, Response response, BinaryContent content)
+      throws MimeTypeException, NullPointerException {
+    String mimeType = content.getMimeTypeValue();
+
+    if (mimeType == null) {
+      LOGGER.debug("Failure to fetch file extension, mime-type is empty");
+      throw new MimeTypeException("Binary Content contains null mime-type value.");
+    }
+
+    String fileExt = getFileExtFromMimeType(mimeType);
+
+    if (containsGzip(request)) {
+      LOGGER.trace("Request header accepts gzip");
+      response.header(HttpHeaders.CONTENT_ENCODING, GZIP);
+    }
+
+    response.type(mimeType);
+    String attachment =
+        String.format("attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+    response.header(HttpHeaders.CONTENT_DISPOSITION, attachment);
+  }
+
+  private String getFileExtFromMimeType(String mimeType) throws MimeTypeException, NullPointerException {
+    MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+    String fileExt = allTypes.forName(mimeType).getExtension();
+    if (fileExt == null || StringUtils.isEmpty(fileExt)) {
+      LOGGER.debug("Fetching file extension from mime-type resulted in null or empty.");
+      throw new NullPointerException("Failure fetching file extension from mime type");
+    }
+    return allTypes.forName(mimeType).getExtension();
+  }
+
+  private Boolean containsGzip(Request request) {
+    return request.headers(HttpHeaders.ACCEPT_ENCODING).toLowerCase().contains(GZIP);
+  }
+
+  private void attachFileToResponse(
+      Request request,
+      Response response,
+      ServiceReference<QueryResponseTransformer> queryResponseTransformer,
+      CqlQueryResponse cqlQueryResponse)
+      throws CatalogTransformerException, IOException, MimeTypeException, NullPointerException {
+    BinaryContent content =
+        bundleContext
+            .getService(queryResponseTransformer)
+            .transform(cqlQueryResponse.getQueryResponse(), Collections.emptyMap());
```

Comment:
❗️ You're sending an empty map as the second parameter to this query response transformer. That means you're not sending information about Hidden Fields, Order, or Aliases.

Please see the existing CsvQueryResponse transformer to see what I mean:

/catalog/transformer/catalog-transformer-csv-queryresponse/src/main/java/ddf/catalog/transformer/csv/CsvQueryResponseTransformer.java@master#L93